plugins/dialog/plugin.js

```diff
@@ -500,6 +503,150 @@ CKEDITOR.DIALOG_STATE_BUSY = 2;
+				var groupId = dialogUiElements[ startIndex ]
+					.getElement()
+					.getAscendant( 'table' )
+					.getId();
+
+				while ( isWithinTheSameRadioGroup( groupId, currentIndex ) ) {
+					currentIndex = updateCurrentIndex( currentIndex, direction );
+				}
+
+				var radioGroup = getRadioGroup( currentIndex, direction );
+
+				if ( radioGroup.length > 0 ) {
+					var checkedItemIndex = getFocusedRadioGroupElement( radioGroup );
+
+					if ( checkedItemIndex === -1 ) {
+						checkedItemIndex = 0;
+						dialogUiElements[ currentIndex ].getInputElement().$.checked = true;
+					}
+
+					currentIndex = updateCurrentIndexBy( currentIndex, checkedItemIndex, direction );
+				}
+			} else {
+				var radioGroupLength = dialogUiElements[ startIndex ].items.length;
+				// When focus is on radio button, Tab or Shift+Tab will exit the radio group or
+				// move to the next radio group.
+				for ( var i = 0; i < radioGroupLength - 1; i++ ) {
+					currentIndex = updateCurrentIndex( currentIndex, direction );
+				}
+			}
+
+			// When the current element is not focusable then depends on direction
+			// get the next or previous element.
+			if ( !dialogUiElements[ currentIndex ].isFocusable() ) {
+				while ( !dialogUiElements[ currentIndex ].isFocusable() ) {
+					currentIndex = updateCurrentIndex( currentIndex, direction );
+				}
+			}
+
+			return currentIndex;
+		}
+
+		function getRadioGroup( startIndex, direction ) {
+			var dialogUiElements = me._.focusList,
+				radioGroup = [],
+				length;
+
+			if ( dialogUiElements[ startIndex ].items ) {
+				length = dialogUiElements[ startIndex ].items.length;
+			}
+
+			if ( length === undefined ) {
+				return radioGroup;
+			}
+
+			for ( var i = 0; i < length; i++ ) {
+				radioGroup.push( dialogUiElements[ startIndex ] );
+				startIndex = updateCurrentIndex( startIndex, direction );
+			}
+
+			return radioGroup;
+		}
+
+		function getFocusedRadioGroupElement( radioGroup ) {
+			return CKEDITOR.tools.indexOf( radioGroup, function( element ) {
+				return element.getInputElement().$.checked;
+			} );
+		}
+
+		function updateCurrentIndex( index, direction ) {
+			var dialogUiElementsLength = me._.focusList.length;
+
+			if ( direction === 'next' ) {
+				return index + 1 < dialogUiElementsLength ? index + 1 : 0;
+			}
+
+			return index - 1 >= 0 ? index - 1 : dialogUiElementsLength - 1;
+		}
+
+		function updateCurrentIndexBy( currentIndex, value, direction ) {
+			if ( direction === 'next' ) {
+				return currentIndex + value;
+			}
+
+			return currentIndex - value;
+		}
+
+		function isWithinTheSameRadioGroup( groupId, index ) {
+			var dialogUiElements = me._.focusList,
+				element;
+
+			// Focusable elements like lock ratio in image2 dialog does not
+			// include getElement() method.
+			try {
+				element = dialogUiElements[ index ].getElement();
+			} catch ( e ) {
+				element = dialogUiElements[ index ].element;
+			}
```

Comment:
In that case, can't it be simply replaced by:

```suggestion
			element = dialogUiElements[ index ].element;
```

?

Reply:
If you replace this code with your suggestion and open `radiogroupfocus` test you'll get an error because method `getElement` does not exist. 
[Here](https://github.com/ckeditor/ckeditor4/blob/t/439/plugins/dialog/plugin.js#L556) we loop through elements till the end of the radio group with one more item before or after the radio group.

So in this test, when you try to focus the previous item by using SHIFT+TAB from the radio group to the "Reset Size" button, you will get an error.
Additionally, radio elements contain a `getElement()` method but no `element` property like Focusable elements and vice versa.